src/v0/destinations/campaign_manager/util.js

```diff
@@ -28,6 +38,96 @@ function convertToMicroseconds(input) {
   return timestamp;
 }
 
+const normalizeEmail = (email) => {
+  const domains = ['@gmail.com', '@googlemail.com'];
+
+  const matchingDomain = domains.find((domain) => email.endsWith(domain));
+
+  if (matchingDomain) {
+    const localPart = email.split('@')[0].replace(/\./g, '');
+    return `${localPart}${matchingDomain}`;
+  }
+
+  return email;
+};
+
+const normalizePhone = (phone, countryCode) => {
+  const phoneNumberObject = parsePhoneNumber(phone, countryCode);
+  if (phoneNumberObject && phoneNumberObject.isValid()) {
+    return phoneNumberObject.format('E.164');
+  }
+  throw new InstrumentationError('Invalid phone number');
+};
+
+// ref:- https://developers.google.com/doubleclick-advertisers/guides/conversions_ec#hashing
+const normalizeAndHash = (key, value, options) => {
+  if (!isDefinedAndNotNull(value)) return value;
+
+  let normalizedValue;
+  const trimmedValue = value.trim().toLowerCase();
+  switch (key) {
+    case 'hashedEmail':
+      normalizedValue = normalizeEmail(trimmedValue);
+      break;
+    case 'hashedPhoneNumber':
+      normalizedValue = normalizePhone(trimmedValue, options.countryCode);
+      break;
+    case 'hashedFirstName':
+    case 'hashedLastName':
+    case 'hashedStreetAddress':
+      normalizedValue = trimmedValue;
+      break;
+    default:
+      return value;
+  }
+  return sha256(normalizedValue);
+};
+
+const prepareUserIdentifiers = (message, isHashingRequired) => {
+  const payload = constructPayload(
+    message,
+    mappingConfig[ConfigCategories.ENHANCED_CONVERSION.name],
+  );
+
+  if (isHashingRequired) {
+    payload.hashedEmail = normalizeAndHash('hashedEmail', payload.hashedEmail);
+    payload.hashedPhoneNumber = normalizeAndHash('hashedPhoneNumber', payload.hashedPhoneNumber, {
+      options: payload.addressInfo.country,
+    });
+    payload.addressInfo.hashedFirstName = normalizeAndHash(
+      'hashedFirstName',
+      payload.addressInfo.hashedFirstName,
+    );
+    payload.addressInfo.hashedLastName = normalizeAndHash(
+      'hashedLastName',
+      payload.addressInfo.hashedLastName,
+    );
+    payload.addressInfo.hashedStreetAddress = normalizeAndHash(
+      'hashedStreetAddress',
+      payload.addressInfo.hashedStreetAddress,
+    );
+  }
```

Comment:
Even if we not hashing we should be normalising or throwing error in case of wrong format

Reply:
Got it